`configs` field is wrongly optional in the type definition

## Repro

Run `tsc` on this `eslint.config.mjs`.

```typescript
import ts from 'typescript-eslint';
import mocha from 'eslint-plugin-mocha';

export default ts.config(
    mocha.configs['recommended'],
);
```

## Expected behavior

Type check should pass.

## Actual behavior

Got the following errors.

```
eslint.config.mjs:10:5 - error TS18048: 'mocha.configs' is possibly 'undefined'.

10     mocha.configs['recommended'],
       ~~~~~~~~~~~~~

eslint.config.mjs:10:5 - error TS2345: Argument of type 'LegacyConfig<RulesRecord, RulesRecord> | Config<RulesRecord> | Config<RulesRecord>[]' is not assignable to parameter of type 'InfiniteDepthConfigWithExtends'.
  Type 'LegacyConfig<RulesRecord, RulesRecord>' is not assignable to type 'InfiniteDepthConfigWithExtends'.
    Type 'LegacyConfig<RulesRecord, RulesRecord>' is not assignable to type 'ConfigWithExtends'.
      Types of property 'extends' are incompatible.
        Type 'string | string[] | undefined' is not assignable to type 'InfiniteDepthConfigWithExtends[] | undefined'.
          Type 'string' is not assignable to type 'InfiniteDepthConfigWithExtends[]'.

10     mocha.configs['recommended'],
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 2 errors in the same file, starting at: eslint.config.mjs:10
```

## Description

These errors were caused because the `mocha` instance is typed as `ESLint.Plugin`.

https://github.com/lo1tuma/eslint-plugin-mocha/blob/1e5a3a1a9597ab54e5cc3d3fc58071009d0335d3/source/plugin.ts#L81

`ESLint.Plugin` is a pure eslint plugin object which knows nothing about eslint-plugin-mocha. So it types `configs` field as optional. However eslint-plugin-mocha actually sets the `configs` field and the field should not be optional.

--- a/source/plugin.test.ts
+++ b/source/plugin.test.ts
@@ -37,7 +37,7 @@ describe('eslint-plugin-mocha', function () {
         const ruleFiles = await determineAllRuleFiles();
 
         for (const file of ruleFiles) {
-            const ruleName = path.basename(file, '.js') as keyof typeof plugin.rules;
+            const ruleName = path.basename(file, '.js');
             assert.ok(ruleName in plugin.rules);
             // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment -- ok
             const importedRuleModule = await import(path.join(rulesDir, file));

--- a/source/plugin.ts
+++ b/source/plugin.ts
@@ -78,6 +78,13 @@ const recommendedRules: Linter.RulesRecord = {
     'mocha/consistent-spacing-between-blocks': 'error'
 };
 
+export type MochaPlugin = ESLint.Plugin & {
+    configs: {
+        all: Linter.Config;
+        recommended: Linter.Config;
+    };
+};
+
 const mochaPlugin = {
     rules: {
         'handle-done-callback': handleDoneCallbackRule,
@@ -121,7 +128,9 @@ const mochaPlugin = {
     }
 } satisfies ESLint.Plugin;
 
-mochaPlugin.configs.all.plugins.mocha = mochaPlugin;
-mochaPlugin.configs.recommended.plugins.mocha = mochaPlugin;
+const plugin: MochaPlugin = mochaPlugin;
 
-export default mochaPlugin;
+plugin.configs.all.plugins.mocha = plugin;
+plugin.configs.recommended.plugins.mocha = plugin;
+
+export default plugin;

--- a/source/plugin.ts
+++ b/source/plugin.ts
@@ -78,57 +78,66 @@ const recommendedRules: Linter.RulesRecord = {
     'mocha/consistent-spacing-between-blocks': 'error'
 };
 
-export type MochaPlugin = ESLint.Plugin & {
-    configs: {
-        all: Linter.Config;
-        recommended: Linter.Config;
+const rules = {
+    'handle-done-callback': handleDoneCallbackRule,
+    'max-top-level-suites': maxTopLevelSuitesRule,
+    'no-async-suite': noAsyncSuiteRule,
+    'no-exclusive-tests': noExclusiveTestsRule,
+    'no-exports': noExportsRule,
+    'no-global-tests': noGlobalTestsRule,
+    'no-hooks': noHooksRule,
+    'no-hooks-for-single-case': noHooksForSingleCaseRule,
+    'no-identical-title': noIdenticalTitleRule,
+    'no-mocha-arrows': noMochaArrowsRule,
+    'no-nested-tests': noNestedTestsRule,
+    'no-pending-tests': noPendingTestsRule,
+    'no-return-and-callback': noReturnAndCallbackRule,
+    'no-return-from-async': noReturnFromAsyncRule,
+    'no-setup-in-describe': noSetupInDescribeRule,
+    'no-sibling-hooks': noSiblingHooksRule,
+    'no-synchronous-tests': noSynchronousTestsRule,
+    'no-top-level-hooks': noTopLevelHooksRule,
+    'prefer-arrow-callback': preferArrowCallbackRule,
+    'consistent-spacing-between-blocks': consistentSpacingBetweenBlocksRule,
+    'consistent-interface': consistentInterfaceRule,
+    'valid-suite-title': validSuiteTitleRule,
+    'valid-test-title': validTestTitleRule,
+    'no-empty-title': noEmptyTitleRule
+};
+
+type MochaConfig = Linter.Config & {
+    plugins: {
+        mocha: ESLint.Plugin;
     };
 };
 
-const mochaPlugin = {
-    rules: {
-        'handle-done-callback': handleDoneCallbackRule,
-        'max-top-level-suites': maxTopLevelSuitesRule,
-        'no-async-suite': noAsyncSuiteRule,
-        'no-exclusive-tests': noExclusiveTestsRule,
-        'no-exports': noExportsRule,
-        'no-global-tests': noGlobalTestsRule,
-        'no-hooks': noHooksRule,
-        'no-hooks-for-single-case': noHooksForSingleCaseRule,
-        'no-identical-title': noIdenticalTitleRule,
-        'no-mocha-arrows': noMochaArrowsRule,
-        'no-nested-tests': noNestedTestsRule,
-        'no-pending-tests': noPendingTestsRule,
-        'no-return-and-callback': noReturnAndCallbackRule,
-        'no-return-from-async': noReturnFromAsyncRule,
-        'no-setup-in-describe': noSetupInDescribeRule,
-        'no-sibling-hooks': noSiblingHooksRule,
-        'no-synchronous-tests': noSynchronousTestsRule,
-        'no-top-level-hooks': noTopLevelHooksRule,
-        'prefer-arrow-callback': preferArrowCallbackRule,
-        'consistent-spacing-between-blocks': consistentSpacingBetweenBlocksRule,
-        'consistent-interface': consistentInterfaceRule,
-        'valid-suite-title': validSuiteTitleRule,
-        'valid-test-title': validTestTitleRule,
-        'no-empty-title': noEmptyTitleRule
+const configs: {
+    all: MochaConfig;
+    recommended: MochaConfig;
+} = {
+    all: {
+        name: 'mocha/all',
+        plugins: { mocha: {} },
+        languageOptions: { globals: globals.mocha },
+        rules: allRules
     },
-    configs: {
-        all: {
-            name: 'mocha/all',
-            plugins: { mocha: {} },
-            languageOptions: { globals: globals.mocha },
-            rules: allRules
-        },
-        recommended: {
-            name: 'mocha/recommended',
-            plugins: { mocha: {} },
-            languageOptions: { globals: globals.mocha },
-            rules: recommendedRules
-        }
+    recommended: {
+        name: 'mocha/recommended',
+        plugins: { mocha: {} },
+        languageOptions: { globals: globals.mocha },
+        rules: recommendedRules
     }
-} satisfies ESLint.Plugin;
+};
 
-const plugin: MochaPlugin = mochaPlugin;
+export type MochaPlugin = ESLint.Plugin & {
+    rules: typeof rules;
+    configs: typeof configs;
+};
+
+const plugin: MochaPlugin = {
+    rules,
+    configs
+};
 
 plugin.configs.all.plugins.mocha = plugin;
 plugin.configs.recommended.plugins.mocha = plugin;

